ee/desktop/runner/runner.go

```diff
@@ -212,40 +231,99 @@ func (r *DesktopUsersProcessesRunner) Interrupt(interruptError error) {
+	// Regardless, we will write the menu data out to a file that can be grabbed by
+	// any desktop user processes, either when they refresh, or when they are spawned.
+	if err := r.writeMenuFile(desktopStatus.Menu); err != nil {
+		return err
+	}
+
+	if desktopStatus.Enabled && r.processSpawningEnabled {
+		// Desktop is currently enabled, and control server wants it to remain enabled
+		// Tell any running desktop user processes that they should refresh the latest status data
+		for uid, proc := range r.uidProcs {
+			client := client.New(r.authToken, proc.socketPath)
+			if err := client.Refresh(); err != nil {
+				level.Error(r.logger).Log(
+					"msg", "error sending refresh command to desktop process",
+					"uid", uid,
+					"pid", proc.process.Pid,
+					"path", proc.path,
+					"err", err,
+				)
+			}
+		}
+		return nil
+	}
+
+	// Desktop may need to be toggled on or off
+	r.toggleDesktop(desktopStatus.Enabled)
+
+	return nil
+}
+
+// toggleDesktop configures the runner to enable/disable desktop, if necessary
+func (r *DesktopUsersProcessesRunner) toggleDesktop(enabled bool) {
+	if !r.processSpawningEnabled && !enabled {
+		// Desktop is currently disabled, and control server wants it to remain disabled
+		return
 	}
 
-	// TODO: TBD how desktop user processes are notified and ingest control data
-	// Below code leverages the HTTP server but I'm thinking it's preferable
-	// to persist the data to a file, and optionally "ping" the server to let it know
-	// an update has occurred. The desktop can read from the file and react on it's own
-	// One advantage here is that the status info can be passed along even if the desktop
-	// server is unavailable (it would read the file on startup)
+	if r.processSpawningEnabled && !enabled {
+		// Desktop is currently enabled, control server is asking us to disable
+		level.Debug(r.logger).Log("msg", "runner is disabling desktop users processes")
+
+		// Stop automatically spawning desktop user processes
+		r.processSpawningEnabled = false
 
-	// for uid, proc := range r.uidProcs {
-	// 	client := client.New(r.authToken, proc.socketPath)
-	// 	if err := client.SetStatus(desktopStatus.Status); err != nil {
-	// 		level.Error(r.logger).Log(
-	// 			"msg", "error sending status command to desktop process",
-	// 			"uid", uid,
-	// 			"pid", proc.process.Pid,
-	// 			"path", proc.path,
-	// 			"err", err,
-	// 		)
-	// 	}
-	// }
+		// Kill any existing desktop user processes, and we're done here
+		r.killDesktopProcesses()
+	}
+
+	if !r.processSpawningEnabled && enabled {
+		// Desktop is currently disabled, control server is asking us to enable
+		level.Debug(r.logger).Log("msg", "runner is enabling desktop users processes")
+
+		// Next execute loop tick will begin spawning desktop user processes
+		r.processSpawningEnabled = true
+	}
+}
+
+// writeMenuFile writes menu data to a shared file for user processes to access
+func (r *DesktopUsersProcessesRunner) writeMenuFile(data menu.MenuData) error {
```

Comment:
Actually, good point, I used to have the desktop enabled attribute in the menu file, but now that it's not, it won't get persisted. I'll have to figure out where to put that piece of data so that the runner can ingest it and obey it upon restart.

Reply:
This is persisted in a `kolide_desktop_flags` bucket now.